src/caencoder.c

```diff
@@ -492,6 +519,49 @@ static int ca_encoder_node_read_dirents(CaEncoderNode *n) {
         return 1;
 }
 
+static int dirent_bsearch_func(const void *key, const void *member) {
+        const char *k = key;
+        const struct dirent ** const m = (const struct dirent ** const) member;
+
+        return strcmp(k, (*m)->d_name);
+}
+
+static int ca_encoder_node_load_exclude_file(CaEncoderNode *n) {
+        int r;
+        assert(n);
+
+        if (!S_ISDIR(n->stat.st_mode))
+                return -ENOTDIR;
+        if (n->fd < 0)
+                return -EBADFD;
+
+        if (n->has_exclude_file == CA_ENCODER_HAS_EXCLUDE_FILE_DONT_KNOW) {
+
+                r = ca_encoder_node_read_dirents(n);
+                if (r < 0)
+                        return r;
+
+                n->has_exclude_file = bsearch(".caexclude", n->dirents, n->n_dirents, sizeof(struct dirent*), dirent_bsearch_func) ?
```

Comment:
An `if` would be clearer, imho.

Reply:
Hmm, i'll rework this slightly to split this in two: the bsearch() expression and the ternary expression